src/lxc/confile.c

```diff
@@ -1790,6 +1792,45 @@ static int set_config_console_logfile(const char *key, const char *value,
 	return set_config_path_item(&lxc_conf->console.log_path, value);
 }
 
+static int set_config_console_logsize(const char *key, const char *value,
+				      struct lxc_conf *lxc_conf, void *data)
+{
+	int pgsz, ret;
+	int64_t size;
+	uint64_t logsize;
+
+	if (lxc_config_value_empty(value)) {
+		lxc_conf->console.log_size = 0;
+		return 0;
+	}
+
+	/* If the user specified "auto" the default log size is 2^17 = 128 Kib */
+	if (!strcmp(value, "auto")) {
+		lxc_conf->console.log_size = 1 << 17;
+		return 0;
+	}
+
+	ret = parse_byte_size_string(value, &size);
+	if (ret < 0)
+		return -1;
+
+	/* Must be at least a page size */
+	pgsz = getpagesize();
+	if (size < pgsz) {
+		ERROR("%s - The size of the console log must be at least %d "
```

Comment:
Would it be better to just automatically set it to the min?

Otherwise users could get annoying start failures when moving a container to a different system.

Reply:
But warn users in the log?

Reply:
addressed